manager/orchestrator/replicated/slot.go

```diff
@@ -27,7 +27,21 @@ func (is slotsByRunningState) Less(i, j int) bool {
 		}
 	}
 
-	return iRunning && !jRunning
+	if iRunning && !jRunning {
+		return true
+	}
+
+	// Use Slot number as a tie-breaker. This will make the order
+	// of task termination deterministic when scaling down.
+	iSlot := is[i]
+	jSlot := is[j]
+	if (iRunning && jRunning) || (!iRunning && !jRunning) {
```

Comment:
As a small cosmetic tweak, could you move this check above as:

```go
if !iRunning && jRunning {
        return false
}
```

This feels a little more symmetric with the `if iRunning && !jRunning` check.

Then as the tie-breaker, we can have `return is[i][0].Slot < is[j][0].Slot`.